cuegui/cuegui/MenuActions.py

```diff
@@ -1046,30 +1074,64 @@ def retry(self, rpcObjects=None):
                 job.retryFrames(name=names)
                 self._update()
 
-    previewMain_info = ["Preview Main", None, "previewMain"]
+    previewMain_info = ["Preview Main", None, "image"]
 
     # pylint: disable=broad-except
-    def previewMain(self, rpcObjects=None):
+    def previewMain(self, rpcObjects=None, katanaMode=False):
         try:
             job = self._getSource()
             frame = self._getOnlyFrameObjects(rpcObjects)[0]
-            d = cuegui.PreviewWidget.PreviewProcessorDialog(job, frame, False)
-            d.process()
-            d.exec_()
+            layer = None
+            layers = job.getLayers()
+            if len(layers) > 0:
+                for ilayer in layers:
+                    if ilayer.name() == frame.layer():
+                        layer = ilayer
```

Comment:
Having an `if len(layers)` followed by a `for i in layers` is redundant. The for will yield nothing is `layers` is empty.

Reply:
I agree. I have a feeling alot of this is copy paste from the old code though :). WIll change it

Reply:
Done